hudi-utilities/src/main/java/org/apache/hudi/utilities/keygen/TimestampBasedKeyGenerator.java

```diff
@@ -35,16 +35,21 @@
 import java.util.Collections;
 import java.util.Date;
 import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 /**
  * Key generator, that relies on timestamps for partitioning field. Still picks record key by name.
  */
 public class TimestampBasedKeyGenerator extends SimpleKeyGenerator {
 
   enum TimestampType implements Serializable {
-    UNIX_TIMESTAMP, DATE_STRING, MIXED, EPOCHMILLISECONDS
```

Comment:
this will break users setting EPOCHMILLISECONDS today.. Could we avoid this pain  and transparently handle this

Reply:
If this is set, we can make the timeunit be MILLISECONDS..

Reply:
yep, will do